controllers/shared.go

```diff
@@ -22,7 +23,21 @@ func nodeUpdateNeedsReconcile(ev event.UpdateEvent) bool {
 	if newNode, ok = ev.ObjectNew.(*v1.Node); !ok {
 		return false
 	}
-	return conditionsNeedReconcile(oldNode.Status.Conditions, newNode.Status.Conditions)
+
+	return labelsUpdateNeedsReconcile(oldNode.Labels, newNode.Labels) ||
+		conditionsNeedReconcile(oldNode.Status.Conditions, newNode.Status.Conditions)
+}
+
+func labelsUpdateNeedsReconcile(oldLabels, newLabels map[string]string) bool {
```

Comment:
nit: `Update` isn't needed in the method name IMHO, and without it aligns to `conditionsNeedReconcile`